held/app/src/main/java/com/held/gcm/GcmIntentService.java

```diff
@@ -90,60 +90,60 @@ private void checkNotificationType(Bundle bundleResponse) {
         Log.i(TAG, " title: " + title);
 
         int gameId, oppToken;
-
-        switch (type) {
-            case "friend:request":
-                int count = PreferenceHelper.getInstance(this).readPreference(getString(R.string.API_FRIEND_REQUEST_COUNT), 0);
-                count++;
-                PreferenceHelper.getInstance(this).writePreference(getString(R.string.API_FRIEND_REQUEST_COUNT), count);
-                Intent intent = new Intent(this, NotificationActivity.class);
-                intent.putExtra("id", 0);
-                sendNotification(intent, title, message);
-
-                break;
-            case "friend:approve":
+if(type!=null) {
```

Comment:
please don't put huge code in conditions. Instead check for other condition eg.
don't do this- 
if (type !== null){
 /// huge code here
}

do this - 
if (type == null){
  // handle error or return
}

// huge code here